_search-plugins/sql/ppl/subsearch.md

```diff
@@ -0,0 +1,244 @@
+
+```sql
+source = outer | where exists [ source = inner1 | where a = c and exists [ source = nested | where c = e ] ]
+source = outer | where exists [ source = inner1 | where a = c | where exists [ source = nested | where c = e ] ]
+```
+
+## `scalar`
+
+A scalar subsearch returns a single value that you can use in comparisons or calculations. This is useful when you need to compare a field against an aggregated value from another query.
+
+### Syntax
+
+```sql
+where <field> = [ search source=... | ... | ... ]
+```
+{% include copy.html %}
+
+### Usage
+
+The following examples demonstrate different ways to implement `scalar` subsearches, from simple aggregation comparisons to complex nested calculations.
+
+#### Uncorrelated
+
+In the following example, the scalar subsearch is independent of the outer query. These subsearches retrieve a single value that can be used in calculations or comparisons:
+
+```sql
+source = outer | eval m = [ source = inner | stats max(c) ] | fields m, a
+source = outer | eval m = [ source = inner | stats max(c) ] + b | fields m, a
+source = outer | where a > [ source = inner | stats min(c) ] | fields a
+source = outer a > [ source = inner | stats min(c) ] | fields a
+```
+{% include copy.html %}
+
+#### Correlated
+
+In the following example, the scalar subsearch references fields from the outer query, creating a dependency where the inner query result depends on each row of the outer query:
+
+```sql
+source = outer | eval m = [ source = inner | where outer.b = inner.d | stats max(c) ] | fields m, a
+source = outer | eval m = [ source = inner | where b = d | stats max(c) ] | fields m, a
+source = outer | eval m = [ source = inner | where outer.b > inner.d | stats max(c) ] | fields m, a
+source = outer | where a = [ source = inner | where outer.b = inner.d | stats max(c) ]
+source = outer | where a = [ source = inner | where b = d | stats max(c) ]
+source = outer | where [ source = inner | where outer.b = inner.d OR inner.d = 1 | stats count() ] > 0 | fields a
+source = outer a = [ source = inner | where b = d | stats max(c) ]
+source = outer [ source = inner | where outer.b = inner.d OR inner.d = 1 | stats count() ] > 0 | fields a
+```
+{% include copy.html %}
+
+#### Nested
+
+The following example demonstrates how to nest multiple scalar subsearches to create complex comparisons or use one subsearch result within another:
+
+```sql
+source = outer | where a = [ source = inner | stats max(c) | sort c ] OR b = [ source = inner | where c = 1 | stats min(d) | sort d ]
+source = outer | where a = [ source = inner | where c =  [ source = nested | stats max(e) by f | sort f ] | stats max(d) by c | sort c | head 1 ]
+```
+{% include copy.html %}
+
+## `relation`
+
+A relation subsearch allows you to use a query result as a dataset in a join operation. This is useful when you need to join with a filtered or transformed dataset rather than joining directly with a static index.
```

Comment:
```suggestion
A `relation` subsearch allows you to use a query result as a dataset in a join operation. This is useful when you need to join with a filtered or transformed dataset rather than joining directly with a static index.
```